squirrel/plugin.py

```diff
@@ -18,30 +18,77 @@ def load_module():
         return importlib.import_module(f'squirrel.plugins.{project_type}')
 
     @staticmethod
-    def get_files(path):
-        find_output = subprocess.run(
-            f'find {path} -type f -not -path "*/[@.]*"',
-            shell=True,
-            capture_output=True,
-            text=True)
-        return find_output.stdout.strip().split('\n')
+    def get_files(wd, ignores):
+        """Function to find all non-ignored files i project directory"""
+        ignore = ignores.get('ignore')
+        project_files = set()
+        for file in iglob('**/*', recursive=True):
+            if path.isfile(file):
+                file_path = path.join(wd, file)
+                project_files.add(file_path)
+        project_files.difference_update(ignore)
+        return project_files
+
+    @staticmethod
+    def import_ignores(wd, file, logger):
+        """Function to read ignore file and store extensions, dir and files
+         to a dictionary"""
+        ignores = {
+            'dir': set(),
+            'file': [],
+            'ignore': set(),
+            }
+        try:
+            with open(file, 'r') as file:
+                tmp_ignore = []
+                tmp_dir = []
+                for line in file.readlines():
+                    add_line = line.strip()
+                    if add_line.startswith('#') or add_line == '':
+                        continue
+                    elif add_line.endswith('/'):
+                        # Add ignored folders and sub folders
+                        # to be passed to Handler()
+                        tmp_dir.extend(
+                            glob(f'{add_line}/**/', recursive=True,))
+                        # Add files inside ignored folder/subfolder
+                        # to be passed to get_files()
+                        tmp_ignore.extend(
+                            glob(f'{add_line}**', recursive=True))
+                    else:
+                        # Add file/extentions to be passed to Handler()
+                        ignores['file'].append(add_line)
+                        # Add all ignored files/extentions
+                        # to be passed to get_files()
+                        tmp_ignore.extend(
+                            glob(f'**/{add_line}', recursive=True))
```

Comment:
Because of `recursive=True`, there's a bug where if I try to ignore a file in the root directory all the files that have the same name get ignored as well which is not the bahaviour expected.

let's take this directory structure as an example:
* test.py
* dir1
  * test.py
```
# ignore file
test.py
```
With the current behaviour, both files get ignored when only the one in the root directory should be. If I wanted to target the other one I would put `dir/test.py`.

Reply:
I tought it should behave as it does now (ignore all files named e.g `test.py` in project folders). But I'll change it so you have to specify `dir/test.py` to ignore inside a sub dir.

Reply:
Now all extensions is ignores recursive
```python
elif add_line.startswith('*'):
    # Add extentions to be passed to Handler()
    ignores['file'].append(add_line)
    # Add all files with current extention
    # to be past to get_files()
    tmp_ignore.extend(
        glob(f'**/{add_line}', recursive=True))
```
and files (e.g `test.py`) is only ignored when in root folder, need to specify `somedir/test.py` in `.squirrelignore` if that one is going to be ignored aswell
```python
else:
    # Add file to be passed to Handler()
    ignores['file'].append(path.join(wd, add_line))
    # Add all ignored files to be passed to get_files()
    tmp_ignore.extend(
        glob(add_line))
```

Reply:
LGTM